coral/src/app/components/ComplexNativeSelect.tsx

```diff
@@ -0,0 +1,88 @@
+import { NativeSelect, NativeSelectProps } from "@aivenio/aquarium";
+import { omit } from "lodash";
+import { ChangeEvent, useState } from "react";
+
+type ComplexNativeSelectProps<ComplexNativeSelectOptionType> =
+  NativeSelectProps & {
+    options: Array<ComplexNativeSelectOptionType>;
+    identifierValue: keyof ComplexNativeSelectOptionType;
+    identifierName: keyof ComplexNativeSelectOptionType;
```

Comment:
For the sake of consistency, we should use the same prop type (`optionToString?: (opt: T) => string;`) to parse option display value as in [Select.tsx](https://github.com/aiven/design-system/blob/main/src/components/Select/Select.tsx). I expect that in the future we would like to support more complex values as simple property picking. E.g. we might want to have the following format `${o.name} - $(o.value)`.